ts/components/Pinpad/KeyPad.tsx

```diff
@@ -132,7 +132,7 @@ const renderPinRow = (
           el.e1,
           el.e2,
           el.e1.isLeft() ? "digit" : "label",
-          `pinpad-digit-${el.e2}`,
+          `pinpad-digit-${el.e1}`,
```

Comment:
This fix actually works but it's not completely safe since the label may be duplicated too, causing the same bug.

What about using the index given by the `map` function?

```suggestion
          `pinpad-digit-${i}`,
```

Reply:
> This fix actually works but it's not completely safe since the label may be duplicated too, causing the same bug.
> 
> What about using the index given by the `map` function?

I knew using index as a key is an anti-pattern, is that okay anyway?

Reply:
yes, you're right it is formerly true, but we need a unique key such as an ID but the Digit object doesn't provide one, the Array index is the only element, in this case that may work as one since this is a static array written through the code.

Do you have any better solution?

Reply:
Compose the key using a number is potentially dangerous.
React has to recognize each element of the VDOM in a unique way.
I guess we can stay pretty sure by composing a key with a specific prefix, like we do in this case

ex: `pinpad-digit-${index}`

Reply:
Ok thank you.. I'll update as suggested @CrisTofani @Undermaken